nntrainer/compiler/previous_input_realizer.cpp

```diff
@@ -9,22 +9,64 @@
  * @author Jihoon Lee <jhoon.it.lee@samsung.com>
  * @bug No known bugs except for NYI items
  */
+#include <algorithm>
 #include <compiler_fwd.h>
 #include <memory>
+#include <stdexcept>
 #include <vector>
 
+#include <layer_node.h>
+#include <nntrainer_log.h>
 #include <previous_input_realizer.h>
 
 namespace nntrainer {
 
 PreviousInputRealizer::PreviousInputRealizer(
-  const std::vector<std::string> &identified_input) {}
+  const std::vector<std::string> &identified_inputs_) :
+  identified_inputs(identified_inputs_) {}
 
 PreviousInputRealizer::~PreviousInputRealizer() {}
 
 GraphRepresentation
 PreviousInputRealizer::realize(const GraphRepresentation &reference) {
-  return GraphRepresentation();
+  GraphRepresentation processed(reference.begin(), reference.end());
+
+  /**
+   * @brief for node has input connection, below function determines if the node
+   * should be input node or add input_layers from previous layer
+   *
+   */
+  auto is_actually_an_input_node = [this](const LayerNode &node) {
+    return node.hasInputShapeProperty() or
+           std::any_of(identified_inputs.begin(), identified_inputs.end(),
+                       [&node](auto &name) { return node.getName() == name; });
+  };
+
+  for (auto iter = processed.begin(); iter != processed.end(); ++iter) {
+    auto &node = *iter;
+    if (node->getNumInputConnections() != 0) {
+      continue;
+    }
+
+    if (is_actually_an_input_node(*node)) {
+      continue;
+    }
+
+    NNTR_THROW_IF(iter == processed.begin(), std::invalid_argument)
+      << "First node must be identified as an input if it is qualified to be "
+         "input, name: "
+      << node->getName();
+
+    auto &prev_node = *(iter - 1);
```

Comment:
ensure its not the first node itself?

Reply:
Could you check if line 55 is the check? if not could you explain more?

Reply:
aah right, my bad.